pkg/operator/csr/cert_controller.go

```diff
@@ -284,6 +284,13 @@ func (c *clientCertificateController) createCSR(ctx context.Context) (string, er
 		},
 	}
 
+	// prevent CSR creation failure on static CSR names
+	if csrStaticName := c.ObjectMeta.Name; len(csrStaticName) > 0 {
```

Comment:
Instead of handling this, I think we should fail early if this is present.  Generated names are better for CSRs than fixed names.

1. CSRs are temporary. They are reaped some time after reaching a terminal state.
2. because they are temporary, it means a fixed name doesn't allow you use existence as a signal
3. fixed names mean that someone else could "steal" your name, leading conflicts.  Generated names do not.

Reply:
Indeed, I agree